internal/view/owner_extender.go

```diff
@@ -0,0 +1,154 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+	"github.com/derailed/k9s/internal/dao"
+	"github.com/derailed/k9s/internal/ui"
+	"github.com/derailed/tcell/v2"
+	"github.com/derailed/tview"
+)
+
+const selectOwnerDialogKey = "owner"
+
+// OwnerExtender adds log actions to a given viewer.
+type OwnerExtender struct {
+	ResourceViewer
+}
+
+// NewOwnerExtender returns a new extender.
+func NewOwnerExtender(v ResourceViewer) ResourceViewer {
+	o := OwnerExtender{
+		ResourceViewer: v,
+	}
+	o.AddBindKeysFn(o.bindKeys)
+
+	return &o
+}
+
+// BindKeys injects new menu actions.
+func (o *OwnerExtender) bindKeys(aa ui.KeyActions) {
+	res, err := dao.AccessorFor(o.App().factory, o.GVR())
+	if err != nil {
+		o.App().Flash().Err(errors.New(fmt.Sprintf("could not get accessor for GVR: %w", err)))
+
+		return
+	}
+	_, ok := res.(dao.Owned)
+	if ok {
+		aa.Add(ui.KeyActions{
+			ui.KeyO: ui.NewKeyAction("Show Owner", o.ownerCmd(), true),
+		})
+	}
+
+}
+
+func (o *OwnerExtender) ownerCmd() func(evt *tcell.EventKey) *tcell.EventKey {
+	return func(evt *tcell.EventKey) *tcell.EventKey {
+		path := o.GetTable().GetSelectedItem()
+		if path == "" {
+			return nil
+		}
+		if !isResourcePath(path) {
+			path = o.GetTable().Path
+		}
+
+		err := o.showOwner(path)
+		if err != nil {
+			o.App().Flash().Err(err)
+		}
+
+		return nil
+	}
+}
+
+func (o *OwnerExtender) showOwner(path string) error {
+	res, err := dao.AccessorFor(o.App().factory, o.GVR())
+	if err != nil {
+		return err
+	}
+	owned, ok := res.(dao.Owned)
+	if !ok {
+		return fmt.Errorf("owner navigation not possible for resource %q", o.GVR())
+	}
+
+	owners, err := owned.GetOwners(path)
+	if err != nil {
+		return err
+	}
+
+	if len(owners) == 0 {
+		return errors.New("resource does not have an owner")
+	}
+
+	if len(owners) == 1 {
+		o.goToOwner(owners[0])
+		return nil
+	}
+
+	return o.showSelectOwnerDialog(owners)
```

Comment:
I think if there are multiple controllers then we should use the same mechanic we currently use for sec/cm aka 'Used By` and show a new view  vs a picker to list out controllers fqn/gvr (please see view.reference.go)
The picker is fine but think it might be more consistent.
What do you think?

Reply:
I wasn't too sure about the picker myself. It's UX is a bit cumbersome for this purpose. But, let's be honest, a resource with multiple owners is rarely seen in the wild. The picker was easy to implement so I went with it. Will look into the view as an alternative if you think it would be required.

Reply:
Yes you are correct! Most instances will be a straight nav. Extending reference view/model, I think is more elegant and could provide more info/actions. 
Given one of the aspect of owner refs is gc, possible this will be more prevalent in crd land.